### DISCUSSION TITLE
Unexpected end of JSON input

### DISCUSSION THREAD

**Discussion by RamaHerbin** [mockoon/mockoon#1178] [Help] [answered]
**Describe the bug**

Hi there,
First of all, I would like to thank you for this fabulous tool which helps me in my daily work.  😊

However I'm encountering a bug when trying to query my data, a list of tickets. Although the GET request correctly returns the list of tickets, an error is displayed in the UI on the desktop Windows application.

The displayed error is "Error while parsing entering body: Unexpected end of JSON input"

Here is the screenshot:
![Capture-mockoon-bug-endOfJSON](https://github.com/mockoon/mockoon/assets/41597427/cd8ad204-c70e-4974-bf8a-faca516e1008)

**To Reproduce**
Here is my configuration:

1. Routes

![Capture-routes](https://github.com/mockoon/mockoon/assets/41597427/bdd65403-e4ff-435a-904c-1709330ea17e)

2. Data

![Capture-data](https://github.com/mockoon/mockoon/assets/41597427/59fcfc02-494b-495d-b36a-eb5793a636c3)

3. Headers

![Capture-eahders2](https://github.com/mockoon/mockoon/assets/41597427/367f1bac-7b0f-421a-b735-150585dda3e3)

**Mockoon version:**
Desktop app v5.1.0

**OS / OS version:**

Windows 10 Entreprise Build 19045.3570

**Comment by 255kb** [comment]
Hi! 

This error message means that the **entering** JSON body is malformed, so the one from the request. You should look at what your app is sending to Mockoon and verify the JSON validity. You can find the entering body in the Logs' request tab. Also, when you click on the "View full body in editor", the JSON parser is enabled and should indicate the error location. 

![image](https://github.com/mockoon/mockoon/assets/7489814/5d9f5590-4c73-42db-af61-695b76b39fb1)

**Reply by RamaHerbin** [reply]
Thanks for your quick rseponse.
Unfortunately I still don't understand, my request body seems to be empty, but the response body is present with valid data.
And why do I need a request body if it's a GET request?


![Capture-bodyResponse](https://github.com/mockoon/mockoon/assets/41597427/fd11de50-88df-49bc-9bb1-208e29f6e502)

![Capture-log](https://github.com/mockoon/mockoon/assets/41597427/e6362b8b-d706-4f47-a6ca-d60e8dd18685)
![Capture-request](https://github.com/mockoon/mockoon/assets/41597427/5654981f-d368-45b6-b593-cfdb330b9506)

**Reply by 255kb** [reply, marked as answer]
You sent `application/json` in the request `Content-Type` header, so Mockoon tries to parse the empty body (which is invalid). Either remove this header from the request or add a minimal valid JSON body (`""`, `[]`, `{}`).
Putting Mockoon aside, it is usually good practice to avoid adding a Content-Type header if you don't send anything in the body. You can read more about it [here](https://stackoverflow.com/questions/29784398/should-content-type-header-be-present-when-the-message-body-is-empty) or [here](https://stackoverflow.com/questions/40614416/is-empty-body-correct-if-content-type-is-application-json).

**Reply by RamaHerbin** [reply]
Understood, thank you !
